lib/generateClass.js

```diff
@@ -549,6 +795,182 @@ function generateFHIRAssignment(cardIsList, baseCardIsList, constraintsLength, f
+            cw.ln(`const sliceMatch = fhir${fhirPathString}.filter(f => ${sliceCriteria});`);
+            cw.bl('if (sliceMatch != null && sliceMatch.length > 0)', () => {
+              cw.ln(`inst.${shrPathString} = inst.${shrPathString} || [];`);
+              cw.ln(`inst.${shrPathString} = inst.${shrPathString}.concat(sliceMatch.map(f => createInstanceFromFHIR('${shrType}', f)));`);
+            });
+          } else {
+            cw.ln(`inst.${shrPathString} = inst.${shrPathString} || [];`);
+            cw.ln(`inst.${shrPathString} = inst.${shrPathString}.concat(fhir${fhirPathString}.map(f => createInstanceFromFHIR('${shrType}', f)));`);
+          }
+        } else if (baseCardIsList) {
+          // not a list in SHR but is a list in FHIR. only keep the matching slice or the first element
+
+          if (sliceCriteria) {
+            cw.ln(`const sliceMatch = fhir${fhirPathString}.find(f => ${sliceCriteria});`);
+            cw.bl('if (sliceMatch != null)', () => {
+              cw.ln(`inst.${shrPathString} = createInstanceFromFHIR('${shrType}', sliceMatch);`);
+            });
+          } else {
+            cw.ln(`inst.${shrPathString} = createInstanceFromFHIR('${shrType}', fhir${fhirPathString}[0]);`);
+          }
+        }
+
+      } else {
+        // If we have to map our elements to an array higher up in the profile
+        if (containerProfileArray && containedPathArray) {
+          // Iterate through each element at current level, and map it to the array element
+          cw.bl(`fhir.${shrElementPath}.forEach( (elem) => `, () => {
+            cw.ln('let containerInst = {};');
+            let containedPathString = '';
+            // Iterate through each level between the array element and our current element, and make sure they're all initialized.
+            containedPathArray.forEach(e => {
+              containedPathString = containedPathString.concat(`['${e}']`);
+              cw.ln(`containerInst${containedPathString} = {};`);
+            });
+            // Set the contained element instance's data with the current 'elem' element
+            cw.ln(`containerInst${containedPathString} = typeof elem.toFHIR === 'function' ? elem.toFHIR() : elem;`);
+            // Push the contained element instance onto the higher-level array
+            cw.ln(`inst${containerProfileArray.map(e => `['${e}']`).join('')}.push(containerInst);`);
+          });
+          // Close the `this.${shrElementPath}.forEach( (elem) => block's parenthesis
+          cw.ln(');');
+        } else {
+          if (shrElementPath == 'value') {
+            cw.ln(`inst.${shrElementPath} = fhir${fhirPathString};`);
+          } else {
+            const shrMapping = shrElementMapping[shrElementMapping.length - 1];
+            const shrType = fhirMappingToIdentifier(shrMapping).fqn;
+            cw.ln(`inst.${shrElementPath} = createInstanceFromFHIR('${shrType}', fhir${fhirPathString});`);
+          }
+        }
+      }
+    }
+  });
+}
```

Comment:
My brain is turning to mush and code generation code is really hard to follow because you need to concurrently be building and remembering the generated code in your head while also reviewing the code that does the generation.  In a way, it's a bit like a meta-code review.

So... I only looked at the code above briefly once I realized my brain was not processing it well anyway.  Instead I'll probably just try it and review the output.